mobile/src/full/java/org/openhab/habdroid/core/GcmIntentService.java

```diff
@@ -64,7 +66,19 @@ protected void onHandleIntent(Intent intent) {
                     notificationId = Integer.parseInt(intent.getExtras().getString(EXTRA_NOTIFICATION_ID));
                 }
                 if ("notification".equals(intent.getExtras().getString("type"))) {
-                    sendNotification(intent.getExtras().getString(EXTRA_MSG), notificationId);
+                    //we use local time as fallback
+                    long timestamp = System.currentTimeMillis();
+                    if(intent.getExtras().containsKey(EXTRA_TIMESTAMP)){
+                        //in case timestamp is passed by cloud service in later versions
+                        timestamp = intent.getExtras().getLong(EXTRA_TIMESTAMP);
+                    }else if(intent.getExtras().containsKey(EXTRA_GCM_TIMESTAMP)){
+                        //use google.sent_time for now as it is the only source
+                        //of the event time available right now
+                        //(or rather the time cloud service forwareded it to gcm)
+                        timestamp = intent.getExtras().getLong(EXTRA_GCM_TIMESTAMP);
+                    }
```

Comment:
Could also simplify this logic:

```
long timestamp = intent.getLongExtra(EXTRA_TIMESTAMP, 0);
if (timestamp == 0) {
    timestamp = intent.getLongExtra(EXTRA_GCM_TIMESTAMP, 0);
}
if (timestamp == 0) {
   timestamp = System.currentTimeMillis();
}
```

or even

```
long timestamp = intent.getLongExtra(EXTRA_TIMESTAMP,
        intent.getLongExtra(EXTRA_GCM_TIMESTAMP, System.currentTimeMillis());
```

The latter certainly deserves a comment, though ;-)